keras_unet/models/custom_unet.py

```diff
@@ -1,75 +1,113 @@
 from keras.models import Model
-from keras.layers import BatchNormalization, Conv2D, Conv2DTranspose, MaxPooling2D, Dropout, UpSampling2D, Input, concatenate
+from keras.layers import (
+    BatchNormalization,
+    Conv2D,
+    Conv2DTranspose,
+    MaxPooling2D,
+    SpatialDropout2D,
+    UpSampling2D,
+    Input,
+    concatenate,
+    Activation,
+)
+
 
 def upsample_conv(filters, kernel_size, strides, padding):
     return Conv2DTranspose(filters, kernel_size, strides=strides, padding=padding)
 
+
 def upsample_simple(filters, kernel_size, strides, padding):
     return UpSampling2D(strides)
 
+
 def conv2d_block(
-    inputs, 
-    use_batch_norm=True, 
-    dropout=0.3, 
-    filters=16, 
-    kernel_size=(3,3), 
-    activation='relu', 
-    kernel_initializer='he_normal', 
-    padding='same'):
-    
-    c = Conv2D(filters, kernel_size, activation=activation, kernel_initializer=kernel_initializer, padding=padding) (inputs)
+    inputs,
+    use_batch_norm=True,
+    dropout=0.3,
+    filters=16,
+    kernel_size=(3, 3),
+    activation="relu",
+    kernel_initializer="he_normal",
+    padding="same",
+):
+
+    c = Conv2D(
+        filters,
+        kernel_size,
+        activation=None,
+        kernel_initializer=kernel_initializer,
+        padding=padding,
+        use_bias=not use_batch_norm,
+    )(inputs)
     if use_batch_norm:
         c = BatchNormalization()(c)
+        c = Activation(activation)(c)
     if dropout > 0.0:
-        c = Dropout(dropout)(c)
-    c = Conv2D(filters, kernel_size, activation=activation, kernel_initializer=kernel_initializer, padding=padding) (c)
+        c = SpatialDropout2D(dropout)(c)
+    c = Conv2D(
+        filters,
+        kernel_size,
+        activation=None,
+        kernel_initializer=kernel_initializer,
+        padding=padding,
+        use_bias=not use_batch_norm,
+    )(c)
     if use_batch_norm:
         c = BatchNormalization()(c)
+        c = Activation(activation)(c)
```

Comment:
@gkaissis I understand that you wanted to move `Activation` layer after the `BatchNormalization` layer but in this case activation layer line shouldn't be placed under `if use_batch_norm` condition. We need to add activation regardless of `use_batch_norm`.
As for `Conv2D->BN->Activation` vs `Conv2D->Activation->BN` can you share some context or research on why exactly you favor the first one over the later? My personal research shows that it either doesn't make any difference or applying BN after activation (e.g. ReLu) makes more sense because we normalize output after removing negative values. Would love to get your perspective on that!

Reply:
Fully agree, this is an indentation issue.
There seems to be no absolute consensus whether BN before activation or after makes a difference (which as you say usually is a sign that it doesn't make a difference). 
That said, there are [some experiments](https://github.com/ducha-aiki/caffenet-benchmark/blob/master/batchnorm.md) stating slightly higher performance with activation after BN, but I don't believe it to be a reproducibly better choice. If you prefer, we can leave it in the layer specification.

>Simple is better than complex.

Reply:
> That said, there are some **experiments stating slightly higher performance with activation after BN**...

Hmm checked the experiment you linked there and it seems like it actually [states the opposite](https://github.com/ducha-aiki/caffenet-benchmark/blob/master/batchnorm.md#bn----before-or-after-relu):

>BN -- before or after ReLU?
>
>Name | Accuracy | LogLoss | Comments
>-- | -- | -- | --
>Before | 0.474 | 2.35 | As in paper
>Before + scale&bias layer | 0.478 | 2.33 | As in paper
>After | 0.499 | 2.21 |  
>After + scale&bias layer | 0.493 | 2.24 |  
>
>**So in all next experiments, BN is put after non-linearity**

According to these experiments BN after non-linearity gives better results.
But nevertheless I agree with you that there seems to be no absolute consensus in this topic..
For now let's keep the activation inside layer specification as it was before

Reply:
Hmm, you are correct. Misread that... Apologies.